### PR TITLE
Clean up mixed license usage

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Address #2887 

I believe this follows the advice that I can find in MM and what I've been told but it's worth a careful check over I think!

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/hdi/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "hdi"
 version = "0.4.0-beta-dev.13"
 description = "The HDI"
-license = "Apache-2.0"
+license-file = "LICENSE_CAL-1.0"
```
Comment (steveej) [inline crates/hdi/Cargo.toml:5]: i think we can use `CAL-1.0` now. when we first started publishing it wasn't available yet but [it is on SPDX now](https://spdx.org/licenses/CAL-1.0.html).

```suggestion
license = "CAL-1.0"
```

File: crates/hdi/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+- Change the license from Apache-2.0 to CAL-1.0 to match the HDK.
```
Comment (steveej) [inline crates/hdi/CHANGELOG.md:10]: i don't remember the reason we chose a different license for the HDI.

@artbrock @thedavidmeister either of you please chime in here.

**Review by steveej — COMMENTED** [review]
thanks for introducing consistency. i think we could drop all the license files at this point, also i think the HDI license is not a mistake and there was a reason for it (which i don't remember and cannot reconstruct).

File: crates/hdi/Cargo.toml (hunk 1 above)
Comment (ThetaSinner) [reply crates/hdi/Cargo.toml]: Good point, updated!

**Review by neonphog — APPROVED** [review]

File: crates/holochain_state/Cargo.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 name = "holochain_state"
 version = "0.3.0-beta-dev.22"
 description = "TODO minimize deps"
```
Comment (maackle) [inline crates/holochain_state/Cargo.toml:4]: Might as well fix this now too

```suggestion
description = "Holochain persisted state datatypes and functions"
```

**Comment by maackle** [comment]
Looks good assuming these are the choices of license that we do want

File: crates/hdi/CHANGELOG.md (hunk 2 above)
Comment (ThetaSinner) [reply crates/hdi/CHANGELOG.md:10]: In order to build a useful zome, you need the HDK so any software you produce is going to be restricted by the CAL anyway. I guess it'd be possible to write a library based on the HDI that is used in integrity zomes only and maybe you'd want a different license for some reason. But the end purpose would still be use in Holochain with the HDK as far as I can imagine. I'm happy to revisit this if necessary but I don't believe it's restrictive to use the CAL license there.

**Pull request merged**